src/main/java/uk/gov/hmcts/ccd/data/userprofile/UserProfileRepository.java

```diff
@@ -39,35 +41,47 @@ public UserProfileRepository(JurisdictionRepository jurisdictionRepository,
         this.userProfileAuditEntityRepository = userProfileAuditEntityRepository;
     }
 
-    /**
-     * Creates a user profile.
-     *
-     * @param userProfile user profile
-     * @param actionedBy for audit trail
-     * @return UserProfile
-     */
     public UserProfile createUserProfile(UserProfile userProfile, final String actionedBy) {
-        if (null != findEntityById(userProfile.getId(), actionedBy, false)) {
-            throw new BadRequestException("User already exists with Id " + userProfile.getId());
-        }
+        AuditAction auditAction;
+        Map<String, JurisdictionEntity> existingJurisdictions = existingJurisdictions(userProfile);
 
-        Map<String, JurisdictionEntity> existingJurisdictions = new HashMap<>();
-        for (Jurisdiction jurisdiction : userProfile.getJurisdictions()) {
-            JurisdictionEntity jurisdictionEntity = jurisdictionRepository.findEntityById(jurisdiction.getId());
-            if (null != jurisdictionEntity) {
-                existingJurisdictions.put(jurisdiction.getId(), jurisdictionEntity);
-            }
+        UserProfileEntity userProfileEntity = findEntityById(userProfile.getId(), actionedBy, false);
+        if (userProfileEntity == null) {
+            auditAction = CREATE;
+            userProfileEntity = UserProfileMapper.modelToEntity(userProfile, existingJurisdictions);
+        } else {
+            auditAction = UPDATE;
```

Comment:
in case of update, can't we reuse the updateUserProfile method?

Reply:
existing updateUserProfile(..) method has workbasket checks inside. If i refactor that too then scope of testing will increase..